lib/trento/infrastructure/software_updates/auth/suma_auth.ex

```diff
@@ -0,0 +1,108 @@
+defmodule Trento.Infrastructure.SoftwareUpdates.Auth.SumaAuth do
+  @moduledoc """
+  GenServer module to authenticate with SUMA
+  """
+
+  use GenServer, restart: :transient
+
+  alias Trento.Infrastructure.SoftwareUpdates.{Suma.State, SumaApi}
+  alias Trento.Infrastructure.SoftwareUpdates.SumaApi
+  alias Trento.SoftwareUpdates
+
+  @default_name "suma_authentication"
+
+  def start_link([]), do: start_link(@default_name)
+
+  def start_link(server_name),
+    do: GenServer.start_link(__MODULE__, %State{}, name: process_identifier(server_name))
+
+  @impl GenServer
+  def init(%State{} = state), do: {:ok, state}
+
+  def identify(server_name \\ @default_name),
+    do:
+      server_name
+      |> identification_tuple
+      |> :global.whereis_name()
+
+  def authenticate(server_name \\ @default_name),
+    do:
+      server_name
+      |> process_identifier
+      |> call(:authenticate)
+
+  def clear(server_name \\ @default_name),
+    do:
+      server_name
+      |> process_identifier
+      |> call(:clear)
+
+  @impl GenServer
+  def handle_call(:authenticate, _, %State{} = state) do
+    case setup_auth(state) do
+      {:ok, new_state} ->
+        {:reply, {:ok, new_state}, new_state}
+
+      {:error, _} = error ->
+        {:reply, error, state}
+    end
+  end
+
+  @impl GenServer
+  def handle_call(:clear, _, _), do: {:reply, :ok, %State{}}
+
+  @impl GenServer
+  def format_status(_reason, [pdict, state]) do
```

Comment:
https://tech.new-work.se/protecting-sensitive-data-in-elixir-genservers-fac4a8b0ae15
They are different things. This is to avoid showing these fields in the tests for example

Reply:
I was sure the protocol implementation was enough. When we implemented it we took inspiration from this post https://gabrielpereira.dev/en/posts/protecting-sensitive-data-in-elixir-gen-servers/

Reply:
here's the PR https://github.com/trento-project/web/pull/2381

Reply:
We posted the same blog at the end hehe
Let me try to understand what it says, because when I read it, maybe I'm getting it wrong.
I understood that both techniques must be implemented to avoid completely the printing of the "secret" data, but I'm assuming now that you say that the `Inspect` protocol fixes the 2 issues at once, without the need for `format_status`, right?

Reply:
yes, `format_status` protects from things like `:sys.get_status` but not `:sys.get_state`.
The inspect protocol implementation from both.

Reply:
But what about the elixir `observer` for example? I'm still trying to figure it out

Reply:
Yes, actually we need the `format_status` to redact things in the observer for example.
Let's keep it, it is a good practice, and we learned something